src/core/opentelemetry.ml

```diff
@@ -815,36 +833,67 @@ module Scope : sig
 
   val links : t -> Span_link.t list
 
+  val status : t -> Span_status.t option
+
   val make :
     trace_id:Trace_id.t ->
     span_id:Span_id.t ->
     ?events:Event.t list ->
     ?attrs:key_value list ->
     ?links:Span_link.t list ->
+    ?status:Span_status.t ->
     unit ->
     t
 
   val to_span_ctx : t -> Span_ctx.t
+  (** Turn the scope into a span context *)
 
   val add_event : t -> (unit -> Event.t) -> unit
+  (** Add an event to the scope. It will be aggregated into the span.
+
+  Note that this takes a function that produces an event, and will only
+  call it if there is an instrumentation backend. *)
 
   val record_exception : t -> exn -> Printexc.raw_backtrace -> unit
 
   val add_attrs : t -> (unit -> key_value list) -> unit
+  (** Add attributes to the scope. It will be aggregated into the span.
+
+  Note that this takes a function that produces attributes, and will only
+  call it if there is an instrumentation backend. *)
 
   val add_links : t -> (unit -> Span_link.t list) -> unit
+  (** Add links to the scope. It will be aggregated into the span.
+
+  Note that this takes a function that produces links, and will only
+  call it if there is an instrumentation backend. *)
+
+  val set_status : t -> Span_status.t -> unit
+  (** set the span status.
+
+  Note that this function will be
+  called only if there is an instrumentation backend. *)
 
   val ambient_scope_key : t Ambient_context.key
+  (** The opaque key necessary to access/set the ambient scope with
+  {!Ambient_context}. *)
 
   val get_ambient_scope : ?scope:t -> unit -> t option
+  (** Obtain current scope from {!Ambient_context}, if available. *)
 
   val with_ambient_scope : t -> (unit -> 'a) -> 'a
+  (** [with_ambient_scope sc thunk] calls [thunk()] in a context where [sc] is
+  the (thread|continuation)-local scope, then reverts to the previous local
+  scope, if any.
+
+  @see <https://github.com/ELLIOTTCABLE/ocaml-ambient-context> ambient-context docs *)
 end = struct
   type item_list =
     | Nil
     | Ev of Event.t * item_list
     | Attr of key_value * item_list
     | Span_link of Span_link.t * item_list
+    | Span_status of Span_status.t * item_list
```

Comment:
Not sure how it matters since that's internal anyway, but this type doesn't enforce the fact that there can only be one `status` per span.

Reply:
I was thinking maybe we can add this info to the `Nil` constructor 

```ocaml
type item_list =
  | Nil of Span_status.t option
  | Ev of Event.t * item_list
  | Attr of key_value * item_list
  | Span_link of Span_link.t * item_list
```

But I'm not sure how it will scale if we add more elements. We may want to switch to a record but then I don't know if that's any better that my initial implementation?

Reply:
I don't think it matters too much that there can be only one :). We can just pick the last one.